Automattic/wp-calypso
# Instant Search: adapt the site input component for both Jetpack and WP.com sites

#### Changes proposed in this Pull Request
Jetpack Search purchase flow for both Jetpack and WP.com sites. 

Changes in experience for Jetpack Search purchase:

* log in to WP.com happens at the start of the flow ( to facilitate site selection )

#### Testing instructions
try logged in/out 
- `/jetpack/connect/search`
- type or paste a site: sites which are listed are WP.com sites and connected Jepack sites.
- follow the steps for Jetpack (with/without active/disconnected) and WP.com sites

![url](https://user-images.githubusercontent.com/13561163/82389205-c958a800-9a3b-11ea-8c1f-2d10eda8810a.png)

Purchase for WP.com sites is not yet implemented, so we keep showing the relevant notice.
